addons/sale/migrations/9.0.1.0/post-migration.py

```diff
@@ -55,6 +55,22 @@ def set_crm_team_message_types(env):
             (4, env.ref('sale.mt_salesteam_invoice_created').id)]})
 
 
+def recalculate_prepopulated_fields(env):
+    """ We prepopulated only lines with uom=product_uom, trigger a recompute
+    for the rest here"""
+    env.cr.execute(
+        """select l.id
+        from sale_order_line l
+        join sale_order_line_invoice_rel r on r.order_line_id=l.id
+        join account_invoice_line il on r.invoice_line_id=il.id
+        where l.product_uom<>il.uom_id
+        """)
+    env['sale.order.line'].invalidate_cache(
+        fnames=['invoice_status', 'qty_invoiced', 'qty_to_invoice'],
+        ids=[x for x, in env.cr.fetchall()]
```

Comment:
the weird syntax is actually slightly faster - compare

`python -m timeit '[x for x, _ in [(y, y) for y in range(100000)]]'`

and

`python -m timeit '[x for x, _ in [(y, y) for y in range(100000)]]'`